spark/spark-3.1/src/test/scala/org/apache/sedona/sql/GeoPackageReaderTest.scala

```diff
@@ -0,0 +1,353 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+package org.apache.sedona.sql
+
+import io.minio.{MakeBucketArgs, MinioClient, PutObjectArgs}
+import org.apache.spark.sql.{DataFrame, SparkSession}
+import org.apache.spark.sql.functions.expr
+import org.apache.spark.sql.sedona_sql.UDT.GeometryUDT
+import org.apache.spark.sql.types.{BinaryType, BooleanType, DateType, DoubleType, IntegerType, StringType, StructField, StructType, TimestampType}
+import org.scalatest.matchers.should.Matchers
+import org.scalatest.prop.TableDrivenPropertyChecks._
+import org.testcontainers.containers.MinIOContainer
+
+import java.io.FileInputStream
+import java.sql.{Date, Timestamp}
+import java.util.TimeZone
+
+class GeoPackageReaderTest extends TestBaseScala with Matchers {
+  TimeZone.setDefault(TimeZone.getTimeZone("UTC"))
+  import sparkSession.implicits._
+
+  val path: String = resourceFolder + "geopackage/example.gpkg"
+  val polygonsPath: String = resourceFolder + "geopackage/features.gpkg"
+  val rasterPath: String = resourceFolder + "geopackage/raster.gpkg"
+  val wktReader = new org.locationtech.jts.io.WKTReader()
+  val wktWriter = new org.locationtech.jts.io.WKTWriter()
+
+  val expectedFeatureSchema = StructType(
+    Seq(
+      StructField("id", IntegerType, true),
+      StructField("geometry", GeometryUDT, true),
+      StructField("text", StringType, true),
+      StructField("real", DoubleType, true),
+      StructField("boolean", BooleanType, true),
+      StructField("blob", BinaryType, true),
+      StructField("integer", IntegerType, true),
+      StructField("text_limited", StringType, true),
+      StructField("blob_limited", BinaryType, true),
+      StructField("date", DateType, true),
+      StructField("datetime", TimestampType, true)))
+
+  describe("Reading geopackage metadata") {
+    it("should read GeoPackage metadata") {
+      val df = sparkSession.read
+        .format("geopackage")
+        .option("showMetadata", "true")
+        .load(path)
+
+      df.where("data_type = 'tiles'").show(false)
```

Comment:
Please remove or comment out all `show()` functions

Reply:
I think we might add linter for that cases, sorry for that.